sources/ClangSharp/Extensions/CXString.cs

```diff
@@ -1,10 +1,24 @@
-﻿using System;
+using System;
 
 namespace ClangSharp
 {
-    public partial struct CXString : IDisposable
+    public unsafe partial struct CXString : IDisposable
     {
-        public string CString => clang.getCString(this);
+        public string CString
+        {
+            get
+            {
+                var pCString = clang.getCString(this);
+
+                if (pCString is null)
+                {
+                    return string.Empty;
+                }
+
+                var span = new ReadOnlySpan<byte>(pCString, int.MaxValue);
+                return span.Slice(0, span.IndexOf((byte)'\0')).AsString();
```

Comment:
Marshaling from native to `string` just utilizes `ROSpan<byte>` and `System.Text.Encoding.UTF8.GetBytes`